torchrl/envs/transforms/r3m.py

```diff
@@ -328,9 +360,36 @@ def device(self):
     def dtype(self):
         return self._dtype
 
+    def set_container(self, container: Union[Transform, EnvBase]) -> None:
+        out = super().set_container(container)
+        # try to get the parent: if it exists, initialize. Otherwise, skip
+        try:
+            parent = self.parent
+            has_parent = parent is not None
+        except AttributeError:
+            has_parent = False
```

Comment:
Maybe you can use the getattr function with a default argument. Something like that:
```python
parent = getattr(self, "parent", None)
has_parent = parent is not None
```

Reply:
thanks for the tip!